File: erpnext/accounts/report/exchange_report/exchange_report.js
// Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt

frappe.query_reports["Exchange Report"] = {
	"filters": [
		{
			"fieldname":"from_date",
			"label": ("From Date"),
			"fieldtype": "Date",
			"width": "80",
			"reqd": 1
		},
		{
			"fieldname":"to_date",
			"label": ("To Date"),
			"fieldtype": "Date",
			"width": "80",
			"reqd": 1
		},

		{
			"fieldname": "currency",
			"label": ("Currency"),
			"fieldtype": "Link",
			"width": "80",
			"options": "Currency"
		},


	]
}

File: erpnext/accounts/report/exchange_report/exchange_report.py
# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

def execute(filters=None):
	columns = get_columns()
	data = get_data(filters)

	return columns, data

def get_columns():
	columns = [
	("Date") + ":date:120",
	("Rate(Nu.)") + ":currency:120",
	]

	return columns

def get_data(filters):
	data  = "select erh.date, erh.exchange_rate FROM `tabExchange Rate History` as erh WHERE docstatus = 1"

	if filters.get("from_date") and filters.get("to_date"):
		data += " and erh.date  between \'" + str(filters.from_date) + "\' and \'"+ str(filters.to_date) + "\'"
	if filters.get("currency"):
		data += " and erh.from_currency = \'" + str(filters.currency) + "\'"
	return frappe.db.sql(data)
